drivers/storage/portworx/component/telemetry.go

```diff
@@ -130,33 +138,30 @@ func (t *telemetry) Reconcile(cluster *corev1.StorageCluster) error {
 	if err := t.setTelemetryCertOwnerRef(cluster, ownerRef); err != nil {
 		return err
 	}
-
-	if pxutil.IsCCMGoSupported(pxutil.GetPortworxVersion(cluster)) {
-		// Remove old ccm components first if exist
-		if err := t.deleteCCMJava(cluster, ownerRef); err != nil {
-			return err
-		}
+	if cluster.Status.ClusterUID == "" {
+		logrus.Warn("clusterUID is empty, wait for it to reconcile telemetry components")
+		return nil
+	}
+	t.isCCMGoSupported = pxutil.IsCCMGoSupported(pxutil.GetPortworxVersion(cluster))
+	if t.isCCMGoSupported {
```

Comment:
nit: This can be a local variable. No need to add it to the receiver.

Reply:
it's used by other functions in ccm java code to determine which service account to use.